src/main/kotlin/dk/cachet/carp/webservices/protocol/repository/CoreProtocolRepository.kt

```diff
@@ -205,4 +206,35 @@ class CoreProtocolRepository(
         return null
     }
 
+    /**
+     * The [getLatestProtocols] function sorts Protocols conditionally by latest Protocols.versionTag
+     * converts a [JsonNode] to a [GetLatestProtocolResponseDto].
+     *
+     * @return A [toLatestProtocolOverview] object containing the ProtocolOverviewDto.
+     */
+
+    suspend fun getLatestProtocols(): List<LatestProtocolOverview?>
+    {
+        val protocols = protocolRepository.findAll()
+        val latestProtocols = mutableListOf<LatestProtocolOverview?>()
+        for (i in protocols)
+        {
+            val protocolSnapshotId = (i.snapshot)!!.firstOrNull()
+            if (protocolSnapshotId != null)
+            {
+                val latestProtocol = getLatestProtocolById(protocolSnapshotId.textValue())
+                if (latestProtocols.none
+                    {
+                        it?.ownerName == latestProtocol?.snapshot?.name &&
+                                it?.snapshot == latestProtocol?.snapshot &&
+                                it?.lastVersionCreatedDate == latestProtocol?.lastVersionCreatedDate
+                    })
+                {
+                    val protocol: LatestProtocolOverview = latestProtocol!!.toLatestProtocolOverview()
+                    latestProtocols += protocol
```

Comment:
cant you just use a set to contain overviews (line 219) and add the overviews created from the latest protocol. The set will then just take care of it if any of them are the same. Then convert back to a list.

Reply:
okey, I think I understand. easy fix